app/models/labware_creators/custom_tagged_plate.rb

```diff
@@ -107,5 +109,100 @@ def create_labware!
+      # created in the Sequencescape API.
+      #
+      # The `transfer_requests_attributes` parameter is an array of hashes,
+      # where each hash represents a transfer request with the following keys:
+      #   - `:source_asset` [String]: The UUID of the source asset (well) from the parent plate.
+      #   - `:target_asset` [String]: The UUID of the target asset (well) in the child plate.
+      #   - `:outer_request` [String]: The UUID of the outer request associated with the transfer.
+      #
+      # The `transfer_requests_attributes` parameter is required when transferring
+      # specific request types from the parent plate to the destination plate. This is
+      # particularly useful in cases where multiple request types exist in the parent plate,
+      # and only a specific type needs to be transferred to the child plate.
+      #
+      # This is utilized within the `transfer_material_from_parent!` method to ensure
+      # that the correct transfer requests are created and grouped for processing.
+      Sequencescape::Api::V2::TransferRequestCollection.create!(
+        transfer_requests_attributes: transfer_requests,
+        user_uuid: user_uuid
+      )
+      true
+    end
+
+    #
+    # Fetches all unique requests associated with a specific well.
+    # This method combines requests where the well is the source and requests associated with aliquots in the well.
+    #
+    # @param [Sequencescape::Api::V2::Well] well The well object for which requests are being fetched.
+    #
+    # @return [Array<Sequencescape::Api::V2::Request>] An array of unique requests associated with the well.
+    #
+    def requests_for_well(well)
+      (well.requests_as_source + well.aliquots.map(&:request)).compact.uniq(&:id)
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    #
+    # Generates transfer requests for the specified child plate.
+    # This method maps each well in the parent plate to its corresponding request and target asset in the child plate.
+    #
+    # @param [Sequencescape::Api::V2::Plate] child_plate The destination plate object for the transfers.
+    #
+    # @return [Array<Hash>] An array of hashes, where each hash represents a transfer request with the following keys:
+    #   - `:source_asset` [String]: The UUID of the source asset (well) from the parent plate.
+    #   - `:outer_request` [String]: The UUID of the outer request associated with the transfer.
+    #   - `:target_asset` [String, nil]: The UUID of the target asset (well) in the child plate, or `nil` if not found.
+    #
+    # @example Example Output
+    #   [
+    #     {
+    #       source_asset: "source-uuid-1",
+    #       outer_request: "request-uuid-1",
+    #       target_asset: "target-uuid-1"
+    #     },
+    #     {
+    #       source_asset: "source-uuid-2",
+    #       outer_request: "request-uuid-2",
+    #       target_asset: "target-uuid-2"
+    #     }
+    #   ]
+    #
+    def create_requests_and_transfers(child_plate)
```

Comment:
I wonder whether all these could be achieved if we made a new WellFIlteredCustomTaggedPlate labware creator by adding WellFilterBehaviour.

Reply:
Thanks for the review @yoldas 
Hmm, I'm not sure. I'm on leave tomorrow (16th May) and @KatyTaylor  on leave next week. Would you mind discussing this with her may be tomorrow (16th May), and see whether it is a better approach. Then I can work on it next week.